docker/Dockerfile.cpu

```diff
@@ -0,0 +1,19 @@
+FROM python:3.11
+
+RUN apt-get update && apt-get install -y \
+	ffmpeg
+
+WORKDIR /app
+
+ADD requirements.txt .
```

Comment:
Remove all the ADD and RUN git clone instead and checkout roop with the next branch

Reply:
I wouldn't recommend this for several reasons:

* When developing locally (without a mount, which is not present in this PR yet) you will have to push your code every time you want to try your changes (and you would have to modify the branch name in the Dockerfile, which should not be git-tracked).
* If you do not use a commit hash as an ARG, you can't get new changes to the branch/repo locally without --no-cache.
* When you build the image in a CI/CD workflow, the code will probably be fetched two times, one time on the runner server (to get the Dockerfile, e.g.) and one time in the RUN step, unnecessarily increasing the build-time.

Reply:
Sorry but I set the requirements for a reason... not asking for a development container but for a production container. It seems you continue to ignore this.

Reply:
As this is not my PR and I just came by to (gladly) see that someone else had already created a Dockerfile for this repo, I am not continuing to ignore the requirements. I just got notified for this thread/PR because I posted in it earlier and wanted to give suggestions based on experience.

Nonetheless, I have now read the requirements and can give a brief explanation why my points are still valid:

> * The Docker container must be able to run without needing to checkout the repository first.

If you create and release a Docker image (with CD workflow, e.g.), users could just directly run the image without having to clone the repo.

> * Roop must be checked out inside Dockerfile to ensure standalone functionality.

Same as above, the code would only need to be checked out in the workflow when building the image.

> * Different docker-compose.yml files can be combined for CPU vs. CUDA version.

This can still be done by just referencing the respective published Docker image in the file(s). Plus, you could even have a `docker-compose.dev.yaml` for contributors, if you wanted to.

> * Use the "docker compose" command without extra shells or other cross-platform incompatible components. [...] An example docker compose command with multiple YAML files is: ```docker-compose -f docker-compose.yml -f docker-compose.cuda.yml up```

Same as above, one could even just do a single `docker run --pull [...]` without having to install docker-compose.

(I intentionally omitted the REST/Web UI/POST parts, because I think they're out of scope for this discussion)

That said, I respect that you are the/a maintainer of this repository and can have your own opinion about this topic. I just wanted to suggest implementing this the way most other projects are doing it (by using a Github CI/CD Workflow, releasing the image to the GHCR), making it easier for people to contribute to and run roop itself. I do not want to force a lenghty discussion or question your PR accpetance rules. :slightly_smiling_face:

Reply:
This is the best practice on building Docker Images in CI Pipelines. In the CI Pipeline you only need to run:

```
docker-compose build
docker-compose push
```

And the images will be published to Docker Hub for public use. The published images are in no way dependent on the Git Repository.

If you want to clone the Git Repository in your CI Pipeline, you will somehow have to tell which commit to checkout or which branch. By building the Docker Images the way it's done in this pull request it happens automatically. The CI Pipeline will build exactly the commit that's been tested and built.

Reply:
Uhm, where did my second comment go? Did you (silently) delete it, @henryruhs?

If not: Sorry for the inconvenience.

If yes: Please refrain from deleting other people's comments without their knowledge if they are not off-topic (I see you found the button for that) or do conflict with the [community guidelines](https://docs.github.com/en/site-policy/github-terms/github-community-guidelines) in any other way. Also, deleting comments just because you don't want to answer them doesn't coincide with  their `Be respectful` point - at least from my perspective. 

For the sake of completeness, here it is again:

--------------- Original comment ---------------

As this is not my PR and I just came by to (gladly) see that someone else had already created a Dockerfile for this repo, I am not continuing to ignore the requirements. I just got notified for this thread/PR because I posted in it earlier and wanted to give suggestions based on experience.

Nonetheless, I have now read the requirements and can give a brief explanation why my points are still valid:

> * The Docker container must be able to run without needing to checkout the repository first.

If you create and release a Docker image (with CD workflow, e.g.), users could just directly run the image without having to clone the repo.

> * Roop must be checked out inside Dockerfile to ensure standalone functionality.

Same as above, the code would only need to be checked out in the workflow when building the image.

> * Different docker-compose.yml files can be combined for CPU vs. CUDA version.

This can still be done by just referencing the respective published Docker image in the file(s). Plus, you could even have a docker-compose.dev.yaml for contributors, if you wanted to.

> * Use the "docker compose" command without extra shells or other cross-platform incompatible components. [...] An example docker compose command with multiple YAML files is: docker-compose -f docker-compose.yml -f docker-compose.cuda.yml up

Same as above, one could even just do a single docker run --pull [...] without having to install docker-compose.

(I intentionally omitted the REST/Web UI/POST parts, because I think they're out of scope for this discussion)

That said, I respect that you are the/a maintainer of this repository and can have your own opinion about this topic. I just wanted to suggest implementing this the way most other projects are doing it (by using a Github CI/CD Workflow, releasing the image to the GHCR), making it easier for people to contribute to and run roop itself. I do not want to force a lenghty discussion or question your PR accpetance rules. :slightly_smiling_face: